### PR TITLE
CHE-746. Split github plugin to several modules client/server/shared

### PR DESCRIPTION
**Pull request by RomanNikitenko**
Signed-off-by: Roman Nikitenko rnikitenko@codenvy.com

### CODE REVIEW

File: plugins/plugin-github/che-plugin-github-server/pom.xml [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            <groupId>org.kohsuke</groupId>
+            <artifactId>github-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
+    </dependencies>
+    <build>
+        <sourceDirectory>src/main/java</sourceDirectory>
```
Comment (sleshchenko) [inline plugins/plugin-github/che-plugin-github-server/pom.xml]: Why do you need it?

File: pom.xml [hunk 2]
```diff
@@ -361,6 +361,16 @@
             </dependency>
             <dependency>
                 <groupId>org.eclipse.che.plugin</groupId>
+                <artifactId>che-plugin-github-server</artifactId>
+                <version>${project.version}</version>
```
Comment (sleshchenko) [inline pom.xml]: remove version

File: plugins/plugin-github/che-plugin-github-server/pom.xml (hunk 1 above)
Comment (RomanNikitenko) [inline plugins/plugin-github/che-plugin-github-server/pom.xml]: We should to write it when we are overriding the default place for source code.
You're right - it isn't this use case, but I prefer to define it clearly.
WDYT?

Comment (sleshchenko) [inline plugins/plugin-github/che-plugin-github-server/pom.xml]: I prefer to remove it but up to you

Comment (RomanNikitenko) [inline plugins/plugin-github/che-plugin-github-server/pom.xml]: I've removed it.

**Comment by vparfonov** [comment]
ok

**Pull request merged**